subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/IndyRestMapperExceptionHandler.java

```diff
@@ -0,0 +1,33 @@
+package org.commonjava.indy.bind.jaxrs;
+
+
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonObjectFormatVisitor;
+import org.commonjava.indy.IndyException;
+import org.commonjava.indy.model.rest.IndyRestMapperResponse;
+import org.jboss.resteasy.client.exception.ResteasyHttpException;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+/*
+    Catch all Server Exceptions propagated back to client caller through HTTP REST calls and returning json object
+    representation of exception message.
+ */
+@Provider
+public class IndyRestMapperExceptionHandler implements ExceptionMapper<Exception> {
+
+
+    @Override
+    public Response toResponse(Exception exception) {
+        IndyRestMapperResponse irmpr = new IndyRestMapperResponse(exception.getMessage(), LocalDateTime.now());
+        Optional
+          .ofNullable(exception.getCause())
+          .ifPresent((exc) -> { irmpr.setCause(exc.getCause()); } );
+        return Response.ok(irmpr, MediaType.APPLICATION_JSON).build();
```

Comment:
We can't use a 200 response here, IMO. This can cause big problems, as we've seen with Sonatype Nexus, which returns 200 OK when it returns errors, and sometimes results in Indy storing an error output as the intended artifact download file.

Reply:
yea , we can change status returned but this is only that server responded ok for request , but Exception is inside json object , so if human eyes are behind request he can see that request has error'ed and didn't return what expected , but if some type of client is getting this result then he must parse returned response for errors or exceptions but either way he will not get raw exception stack tree.
So it's not difficult to change to returned bad state response  but we must decide if we want that.

Reply:
As im thinking deeper maybe returning some type of specific "bad" response status code is better solution...  My concern is that if we return "ok" status code then for all our previous implemented code that will mean that everything is ok , and will not check for exceptions , so if that is the case we must add checkers or ask for them in every http request...

Reply:
Generally 400-series responses are "it's your fault" while 500-series codes are "oops, that's my fault" from the server. 200-series are various forms of okay, and 300s are "look somewhere else for this." I think if it's an exception and not IndyWorkflowException, then it should be 500 response, and let the user read the body for specifics. If it is IndyWorkflowException, then we should look to see what error code is embedded.

Reply:
That is my perspective also , problem here ( its not a general problem , it is more of design perspective ) is that we are having access only to happened exception object and not status code from response , real use case for this kind of exception mapping is throwing specialized exceptions and catch them here and gathering all bad status code based on that assumption - >=400 < 500 -bad client , >=500 < 600 - bad server.

Reply:
@jdcasey i have made reduction on exceptions based on their types . What you think? is it ok like this?